plugins/woocommerce/includes/admin/settings/class-wc-settings-accounts.php

```diff
@@ -278,29 +330,45 @@ public function output() {
 		?>
 		<script type="text/javascript">
 			document.addEventListener('DOMContentLoaded', function() {
+				// Move tooltips to label element. This is not possible through the settings field API so this is a workaround
+				// until said API is refactored.
+				document.querySelectorAll('input[disabled-tooltip]').forEach(function(element) {
+					const label = element.closest('label');
+					label.setAttribute('disabled-tooltip', element.getAttribute('disabled-tooltip'));
+				});
+
+				// This handles settings that are enabled/disabled based on other settings.
 				const checkboxes = [
 					document.getElementById("woocommerce_enable_signup_and_login_from_checkout"),
 					document.getElementById("woocommerce_enable_myaccount_registration"),
+					document.getElementById("woocommerce_enable_delayed_account_creation"),
 					document.getElementById("woocommerce_enable_signup_from_checkout_for_subscriptions")
 				];
 				const inputs = [
 					document.getElementById("woocommerce_registration_generate_username"),
 					document.getElementById("woocommerce_registration_generate_password")
 				];
-
-				function updateInputs() {
+				checkboxes.forEach(cb => cb && cb.addEventListener('change', function() {
 					const isChecked = checkboxes.some(cb => cb && cb.checked);
 					inputs.forEach(input => {
 						if ( ! input ) {
 							return;
 						}
 						input.disabled = !isChecked;
-						input.closest('td').classList.toggle("disabled", !isChecked);
 					});
-				}
+				}));
+				checkboxes[0].dispatchEvent(new Event('change')); // Initial state
+
+				// Guest checkout should toggle off some options.
+				const guestCheckout = document.getElementById("woocommerce_enable_guest_checkout");
 
-				checkboxes.forEach(cb => cb && cb.addEventListener('change', updateInputs));
-				updateInputs(); // Initial state
+				if ( guestCheckout ) {
+					guestCheckout.addEventListener('change', function() {
+						const isChecked = this.checked;
+						document.getElementById("woocommerce_enable_delayed_account_creation").disabled = !isChecked;
```

Comment:
If the flag is off, this will throw an error trying to find the element:
```
Uncaught TypeError: Cannot set properties of null (setting 'disabled')
    at HTMLInputElement.<anonymous> (admin.php?page=wc-settings&tab=account:1023:87)
```

Furthermore it seems to spawn a jQuery migrate warning :/
```
jquery-migrate.js?ver=3.4.1:136 JQMIGRATE: jQuery.fn.removeAttr no longer sets boolean properties: disabled`
```

Reply:
The jQuery one is coming from `$( '.woocommerce-save-button' ).removeAttr( 'disabled' );`. This code is not jQuery.

Reply:
Other is fixed in https://github.com/woocommerce/woocommerce/pull/51236/commits/ef0ca6983271c0d6bb845be474045aaeaeb2a350